src/main/webapp/js/common/react/tests/Navbar.test.js

```diff
@@ -2,59 +2,86 @@ import React from 'react';
 import Navbar from '../components/Navbar.js';
 import '@testing-library/jest-dom/extend-expect';
 import {render, screen} from '@testing-library/react';
-import fetchMock from 'fetch-mock-jest';
-import {makeRelativeUrlAbsolute} from '../../../fetch_handler.js';
 
 const navbarUrls = [
   {
     href: '/',
     name: 'Home',
+    requiresAuth: false,
   },
   {
     href: '/projects.html',
     name: 'Projects',
+    requiresAuth: false,
+  },
+  {
+    href: '/dashboard.html',
+    name: 'Dashboard',
+    requiresAuth: true,
   },
 ];
 
-beforeAll(() => {
-  const authData = {
-    authorized: false,
-    loginUrl: '/login',
-    logoutUrl: '/logout',
-    hasProfile: false,
-  };
-
-  fetchMock.get(makeRelativeUrlAbsolute('/auth'), {
-    body: JSON.stringify(authData),
-    status: 200,
-    headers: {
-      'Content-Type': 'application/json',
-    },
-  });
-});
-
-afterAll(() => {
-  fetchMock.restore();
-});
+const mockAuthData = {
+  authorized: true,
+  hasProfile: true,
+};
 
 describe('Navbar component', () => {
   it('is added to the DOM', () => {
-    const elem = render(<Navbar urls={navbarUrls} />);
+    const elem = render(
+        <Navbar urls={navbarUrls} loading={true} />);
     expect(elem.container.firstChild).not.toBeNull();
   });
 
   it('displays the names of the urls', () => {
-    render(<Navbar urls={navbarUrls} />);
+    render(
+        <Navbar urls={navbarUrls} loading={false} authData={mockAuthData} />);
 
     expect(screen.queryByText('Home')).not.toBeNull();
     expect(screen.queryByText('Projects')).not.toBeNull();
+    expect(screen.queryByText('Dashboard')).not.toBeNull();
   });
 
-  it('contains both urls', () => {
-    render(<Navbar urls={navbarUrls} />);
+  it('contains the urls', () => {
+    render(
+        <Navbar urls={navbarUrls} loading={false} authData={mockAuthData} />);
 
     expect(screen.getByText('Home')).toHaveAttribute('href', '/');
     expect(screen.getByText('Projects'))
         .toHaveAttribute('href', '/projects.html');
+    expect(screen.getByText('Dashboard'))
+        .toHaveAttribute('href', '/dashboard.html');
+  });
+
+  it('does not display a restricted link to an unauthenticated user', () => {
+    const authData = {
+      authorized: false,
+      hasProfile: false,
+    };
+
+    render(
+        <Navbar urls={navbarUrls} loading={false} authData={authData} />);
+
+    expect(screen.queryByText('Dashboard')).toBeNull();
+  });
+
+  it('does not display a restricted link to a user without a profile', () => {
+    const authData = {
+      authorized: true,
+      hasProfile: false,
+    };
+
+    render(
+        <Navbar urls={navbarUrls} loading={false} authData={authData} />);
+
+    expect(screen.queryByText('Dashboard')).toBeNull();
+  });
+
+  it('does not display the restricted link while loading auth data', () => {
+    render(<Navbar urls={navbarUrls} loading={true} />);
+
+    expect(screen.queryByText('Home')).not.toBeNull();
```

Comment:
Here you explicitely check that 'Home' and 'Projects' are not null, but you do not check this for the unauthenticated or profile-less tests. I'm just wondering why.

Reply:
That's a good point, and seems to be a bit of an inconsistency. I'll update that now.